module/VuFind/src/VuFind/ILS/Connection.php

```diff
@@ -973,6 +982,42 @@ public function getMyTransactions($patron, $params = [])
         return $result;
     }
 
+    /**
+     * Get holdings
+     *
+     * Retrieve holdings from ILS driver class and normalize result array if needed.
+     *
+     * @param string $id      The record id to retrieve the holdings for
+     * @param array  $patron  Patron data
+     * @param array  $options Possible additional options
+     *
+     * @return array           Array with holding data
+     */
+    public function getHolding($id, $patron = null, $options = null)
+    {
+        $result = $this->__call('getHolding', [$id, $patron, $options]);
+
+        // Return only "holdings" if the result contains a top level array with
+        // a "total" key
+        if (isset($result['total'])) {
+            $this->totalItemCount = $result['total'];
+            return $result['holdings'];
+        }
+
+        $this->totalItemCount = count($result);
+        return $result;
+    }
+
+    /**
+     * Get the total number of items for a given bib id, returned from ILS
+     *
+     * @return int Total number of items returned from ILS
+     */
+    public function getTotalItemCount()
```

Comment:
I'm glad that you've moved getTotalItemCount out of the Alma driver, but I hadn't expected to see it as part of the Connection class either. I was actually thinking more along the lines of having getHolding return the new format from the Alma driver, and back-fill from legacy drivers, like:

<pre>
    public function getHolding($id, $patron = null, $options = null)
    {
        $result = $this->__call('getHolding', [$id, $patron, $options]);
        return isset($result['total']) ? $result : ['total' => count($result), 'holdings' => $result];
    }
</pre>

...and then, of course, all the code that calls getHolding would need to be adjusted... but I don't think that's a large number of places. If you would like me to work on this implementation as a separate pull request, please let me know and I'm happy to take a little time to do so!

Reply:
OK, it seems that I thought the other way round: Defaulting the new return format of the Alma driver so that no other code needs to be touched. But I changed that now. You were right: There were not much places where it needed to be touched. I also could get rid of getTotalItemCount by returning the total with the getHolding result.

Reply:
Great, it sounds like we're on the same page now. :-)